src/Optimizely/Config/DatafileProjectConfig.php

```diff
@@ -230,19 +273,19 @@ public function __construct($datafile, $logger, $errorHandler)
 
         $this->_accountId = $config['accountId'];
         $this->_projectId = $config['projectId'];
-        $this->_anonymizeIP = isset($config['anonymizeIP'])? $config['anonymizeIP'] : false;
-        $this->_botFiltering = isset($config['botFiltering'])? $config['botFiltering'] : null;
+        $this->attributes = $config['attributes'] ?: [];
+        $this->audiences = $config['audiences'] ?: [];
+        $this->events = $config['events'] ?: [];
+        $this->typedAudiences = isset($config['typedAudiences']) ? $config['typedAudiences'] : [];
+        $this->_anonymizeIP = isset($config['anonymizeIP']) ? $config['anonymizeIP'] : false;
+        $this->_botFiltering = isset($config['botFiltering']) ? $config['botFiltering'] : null;
         $this->_revision = $config['revision'];
         $this->_sendFlagDecisions = isset($config['sendFlagDecisions']) ? $config['sendFlagDecisions'] : false;
 
         $groups = $config['groups'] ?: [];
         $experiments = $config['experiments'] ?: [];
-        $events = $config['events'] ?: [];
```

Comment:
any reason to remove it?

Reply:
they were being used as variables but i need to get them as attributes so delete the variables and using attributes in further functions